android-components/components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt

```diff
@@ -52,6 +55,7 @@ class ReaderViewFeature(
     private val engine: Engine,
     private val store: BrowserStore,
     controlsView: ReaderViewControlsView,
+    private val createUUID: UUIDCreator = { UUID.randomUUID().toString() },
```

Comment:
I also got this mixed up yesterday and @boek had to correct me. The short of this is that this is not a persistent cache and is really just a tmp cache to try and beat the race condition with the network request for the page. 

Because we're moving to readerview with the currently loaded page we can serialize it to disk and deserialize to show the page in Readerview mode but we also fire off a network request for the page. The race condition is on purpose to increase performance and on most newer phones the serialize/deserialize wins over the network request.

This UUID will change after every open and close of the readermode (this tripped me up) and is not meant to be used for any lifecycle changes as far as I'm aware.

Reply:
Oops, I did not read [the MDN doc](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/session) properly. 

> Represents the session storage area. Items in session storage are stored in memory for the duration of the browser session and are not persisted to disk.

Thanks for clarifying Zac!

> Because we're moving to readerview with the currently loaded page we can serialize it to disk and deserialize to show the page in Readerview mode but we also fire off a network request for the page. The race condition is on purpose to increase performance and on most newer phones the serialize/deserialize wins over the network request.

Yeah! This is based on the RCWN code that I mentioned in our channel.